digdag-storage-gcs/src/main/java/io/digdag/storage/gcs/GCSStorageFactory.java

```diff
@@ -0,0 +1,50 @@
+package io.digdag.storage.gcs;
+
+import java.io.IOException;
+import java.io.FileInputStream;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.storage.StorageOptions;
+import io.digdag.client.config.Config;
+import io.digdag.client.config.ConfigException;
+import io.digdag.spi.Storage;
+import io.digdag.spi.StorageFactory;
+
+public class GCSStorageFactory
+    implements StorageFactory
+{
+    @Override
+    public String getType()
+    {
+        return "gcs";
+    }
+
+    @Override
+    public Storage newStorage(Config config)
+    {
+
+        com.google.cloud.storage.Storage storage;
+        if (config.has("creadentials.json_key")) {
```

Comment:
Fix typo. s/creadentials/credentials/

Reply:
`credentials.json` may be more commonly used term, or `credentials.file` may be good as s3 storage uses.

Reply:
Supporting a text instead of a file may be useful when we can not mount or copy credentials.file to k8s or hadoop nodes easily.

Some of embulk plugin supports `content` https://github.com/embulk/embulk-output-gcs#json-key-of-gcps-service-account. s3 storage seems to support directly specifying values.